Description: fix build failure on GNU/kFreeBSD
 We only care for kernel not the entire FreeBSD userland.
 Update the ./configure check so we set the correct default serial
 and parallel on kfreebsd.
Author: Aurelien Jarno <aurel32@debian.org>
Debian-Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=329448
Index: avrdude/ppi.c
===================================================================
--- avrdude.orig/ppi.c	2010-01-17 02:51:50.814484015 +0100
+++ avrdude/ppi.c	2010-01-17 13:10:29.833688002 +0100
@@ -33,7 +33,7 @@
 #include <unistd.h>
 #include <errno.h>
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 # include "freebsd_ppi.h"
 #elif defined(__linux__)
 # include "linux_ppdev.h"
Index: avrdude/par.c
===================================================================
--- avrdude.orig/par.c	2010-01-17 02:51:50.806489420 +0100
+++ avrdude/par.c	2010-01-17 13:10:29.833688002 +0100
@@ -28,7 +28,7 @@
 #include <unistd.h>
 #include <errno.h>
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 # include "freebsd_ppi.h"
 #elif defined(__linux__)
 # include "linux_ppdev.h"
Index: avrdude/configure.ac
===================================================================
--- avrdude.orig/configure.ac	2010-01-17 02:51:50.774483106 +0100
+++ avrdude/configure.ac	2010-01-17 13:10:54.701501874 +0100
@@ -196,11 +196,11 @@
 		DEFAULT_PAR_PORT="unknown"
 		DEFAULT_SER_PORT="/dev/ttyS0"
 		;;
-	i[[3456]]86-*-freebsd*|amd64-*-freebsd*)
+	i[[3456]]86-*-*freebsd*|amd64-*-*freebsd*)
 		DEFAULT_PAR_PORT="/dev/ppi0"
 		DEFAULT_SER_PORT="/dev/cuaa0"
 		;;
-	*-*-freebsd*)
+	*-*-*freebsd*)
 		DEFAULT_PAR_PORT="unknown"
 		DEFAULT_SER_PORT="/dev/cuaa0"
 		;;
